app/streaming/video/ffmpeg-renderers/vaapi.cpp

```diff
@@ -442,3 +449,106 @@ VAAPIRenderer::renderFrame(AVFrame* frame)
         SDL_assert(false);
     }
 }
+
+#ifdef HAVE_EGL
+
+bool
+VAAPIRenderer::canExportEGL() {
+    return true;
+}
+
+bool
+VAAPIRenderer::initializeEGL(EGLDisplay,
+                             const EGLExtensions &ext) {
+    if (!ext.is_supported("EGL_EXT_image_dma_buf_import")) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION,
+                     "VAAPI-EGL: DMABUF unsupported...");
+        return false;
+    }
+    m_egl_ext_dmabuf = ext.is_supported("EGL_EXT_image_dma_buf_import_modifiers");
+    return true;
+}
+
+ssize_t
+VAAPIRenderer::exportEGLImages(AVFrame *frame, EGLDisplay dpy,
+                               EGLImage images[EGL_MAX_PLANES]) {
+    ssize_t count = 0;
+    auto hwFrameCtx = (AVHWFramesContext*)frame->hw_frames_ctx->data;
+    AVVAAPIDeviceContext* vaDeviceContext = (AVVAAPIDeviceContext*)hwFrameCtx->device_ctx->hwctx;
+
+    VASurfaceID surface_id = (VASurfaceID)(uintptr_t)frame->data[3];
+    VAStatus st = vaExportSurfaceHandle(vaDeviceContext->display,
+                                        surface_id,
+                                        VA_SURFACE_ATTRIB_MEM_TYPE_DRM_PRIME_2,
+                                        VA_EXPORT_SURFACE_READ_ONLY | VA_EXPORT_SURFACE_SEPARATE_LAYERS,
+                                        &m_descriptor);
+    if (st != VA_STATUS_SUCCESS) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION,
+                     "vaExportSurfaceHandle failed: %d", st);
+        return -1;
+    }
+
+    SDL_assert(m_descriptor.num_layers <= EGL_MAX_PLANES);
+
+    st = vaSyncSurface(vaDeviceContext->display, surface_id);
+    if (st != VA_STATUS_SUCCESS) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION,
+                     "vaSyncSurface failed: %d", st);
+        goto sync_fail;
+    }
+
+    for (size_t i = 0; i < m_descriptor.num_layers; ++i) {
+        const auto &layer = m_descriptor.layers[i];
+        const auto &object = m_descriptor.objects[layer.object_index[0]];
+
+        EGLAttrib attribs[17] = {
+            EGL_LINUX_DRM_FOURCC_EXT, (EGLint)layer.drm_format,
+            EGL_WIDTH, i == 0 ? frame->width : frame->width / 2,
+            EGL_HEIGHT, i == 0 ? frame->height : frame->height / 2,
+            EGL_DMA_BUF_PLANE0_FD_EXT, object.fd,
+            EGL_DMA_BUF_PLANE0_OFFSET_EXT, (EGLint)layer.offset[0],
+            EGL_DMA_BUF_PLANE0_PITCH_EXT, (EGLint)layer.pitch[0],
+            EGL_NONE,
+        };
+        if (m_egl_ext_dmabuf) {
```

Comment:
What's the impact if `EGL_EXT_image_dma_buf_import_modifiers` is not supported?

Reply:
To be honest I am not sure (I don't have anything to test it) but  most DMA-BUF user I found have this fallback (wlroots, weston, kodi).

PS: the renderer still works on my computer if I disable this code